### PR TITLE
chore(suite): Add error rule for not overriding components and product-components

### PR DESCRIPTION
**Pull request by jvaclavik**
## Description

- overriding components is now throwing error, not just warning
- all existing occurrences have comment for disabling this rule as exception (it's going to be refactored later)
- add rule also for product-components

This allows not seeing 300+ warnings during checking every PR

## Related Issue

Resolve

### CODE REVIEW

File: packages/components/src/index.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export * from './components/Icon/Icon';
 export * from './components/buttons/PinButton/PinButton';
 export * from './components/buttons/TextButton/TextButton';
 export * from './components/buttons/TooltipButton/TooltipButton';
+export * from './components/Table/Table';
```
Comment (peter-sanderson) [inline packages/components/src/index.ts]: you wanted to not to do `*` exports maybe lets not adding a new one

File: packages/transport-bridge/src/ui/components/Card.tsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import styled from 'styled-components';
 
-import { Card as ComponentsCard } from '@trezor/components/src/components/Card/Card';
+import { Card as ComponentsCard } from '@trezor/components';
```
Comment (peter-sanderson) [inline packages/transport-bridge/src/ui/components/Card.tsx:3]: :clap:

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/CoinmarketTermsModal.tsx [hunk 3]
```diff
@@ -123,7 +123,7 @@ export const CoinmarketTermsModal = ({
                 </Row>
                 <Row alignItems="center">
                     <IconWrapper $backgroundColor="#e8f3fa">
-                        <Icon size={24} name={`checkActive`} color="#1d88c5" />
```
Comment (peter-sanderson) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/CoinmarketTermsModal.tsx:126]: It is interesting that this is not covered by current eslint rules. Can it be?

**Review by peter-sanderson — APPROVED** [review]
In general I really like the idea of ignoring all the current occurrences and fixing them later without bugging devs with ton of warnings.

But it shall be done rather sooner then later, as this `// ignore` comment can be easily taken as an inspiration for new overrides :smile:

File: packages/components/src/index.ts (hunk 1 above)
Comment (jvaclavik) [reply packages/components/src/index.ts]: This shouldn't be in this PR 🫣 deleting

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/CoinmarketTermsModal.tsx (hunk 3 above)
Comment (jvaclavik) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/CoinmarketTermsModal.tsx:126]: It should be definitely possible to set it in prettier and auto correct it.

Comment (peter-sanderson) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/CoinmarketTermsModal.tsx:126]: I think its already there, but probably not working / not configured for backticks ``

**Pull request merged**